js/ui/control/scale.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+var util = require('../../util/util');
+var Control = require('./control');
+var DOM = require('../../util/dom');
+
+module.exports = Scale;
+
+function Scale(options) {
+    util.setOptions(this, options);
+}
+
+Scale.prototype = util.inherit(Control, {
+    options: {
+    position: 'bottom-left'
+    },
+
+    onAdd: function(map) {
+        var className = 'mapboxgl-ctrl';
+
+        var container = this._container = DOM.create('div', className + '-group', map.getContainer());
+
+        this._scale = DOM.create('div', ('mapboxgl-ctrl-scale'), this._container);
+
+        _updateScale(map, container);
+         map.on('moveend', function() {
+            _updateScale(map, container);
+         });        
+
+        return container;
+    }
+});
+
+function _updateScale(map, scale) {
```

Comment:
We use `_` prefixes in Javascript to indicate to other parts of the system that a method is private. Because these functions are hidden from other parts of the system because of their file scope, no underscore is needed.